### PR TITLE
Add Description for Feature classes

### PR DESCRIPTION
**Pull request by gastaldi**
Follow up on https://github.com/quarkusio/quarkus/pull/26943

This produces an output during the GraalVM compilation explaining what the Feature does and allows you to browse to the existing source code:

https://user-images.githubusercontent.com/54133/182875747-559e7947-7123-4ff6-95fa-8def09fef6af.mov

### CODE REVIEW

**Review by yrodiere — COMMENTED** [review]
This is nice, but shouldn't the links be something more stable? These will break as soon as we change the directory structure in our repo, and that kind of change *has* been discussed recently.

Maybe a link to the javadoc (which would only depend on the package/class name) would be safer? I couldn't find where the javadoc of Quarkus is published, though...

**Comment by gastaldi** [comment]
> This is nice, but shouldn't the links be something more stable? These will break as soon as we change the directory structure in our repo, and that kind of change _has_ been discussed recently.

We could point to a tagged version instead (by having the URL changed during the build) but I am not sure it's worth the effort. We can start by providing only the descriptions now.

**Comment by geoand** [comment]
Very interesting!

But I totally agree with @yrodiere that we should add something more stable as the URL.

**Comment by gastaldi** [comment]
I reworked the PR to provide only the description for now, until we figure out a stable URL to use

File: core/runtime/src/main/java/io/quarkus/runtime/graal/ResourcesFeature.java [hunk 1]
```diff
@@ -14,16 +15,23 @@ public class ResourcesFeature implements Feature {
...
     @Override
     public void beforeAnalysis(BeforeAnalysisAccess access) {
-        try (BufferedReader in = new BufferedReader(new InputStreamReader(
-                getClass().getClassLoader().getResourceAsStream(META_INF_QUARKUS_NATIVE_RESOURCES_TXT)))) {
-            String line;
-            while ((line = in.readLine()) != null) {
-                if (!line.isEmpty()) {
-                    ResourceHelper.registerResources(line);
+        InputStream resourceAsStream = getClass().getClassLoader().getResourceAsStream(META_INF_QUARKUS_NATIVE_RESOURCES_TXT);
+        if (resourceAsStream != null) {
```
Comment (yrodiere) [inline core/runtime/src/main/java/io/quarkus/runtime/graal/ResourcesFeature.java:19]: I guess you're also fixing a bug here?

That was unexpected, but ok.

**Review by yrodiere — APPROVED** [review]
LGTM, thanks

**Pull request merged**